site/src/hooks/usePaginatedQuery.test.ts

```diff
@@ -0,0 +1,407 @@
+import { renderHookWithAuth } from "testHelpers/renderHelpers";
+import { waitFor } from "@testing-library/react";
+
+import {
+  type PaginatedData,
+  type UsePaginatedQueryOptions,
+  usePaginatedQuery,
+} from "./usePaginatedQuery";
+
+beforeAll(() => {
+  jest.useFakeTimers();
+});
+
+afterAll(() => {
+  jest.useRealTimers();
+  jest.clearAllMocks();
+});
+
+function render<
```

Comment:
hooks don't really get "rendered" so to speak, so this name feels a bit weird. maybe it could be called `testPaginatedQuery` or something? not a huge deal tho